### ISSUE TITLE
Messages are not loading properly

### ISSUE THREAD

**Issue by SerekKiri** [zulip/zulip-mobile#3172]
I'm using **xiaomi redmi 4x** with miui **9.6.5.0** for zulip. When I enter the app, I can see notifications that someone send something on one of the topic but when I enter this topic, messages are not loading. I need to click *Mark as read*, after that they're loading properly (It's happening everytime, not just sometimes and I know this issue is not only on my phone).

I can't find issue similar to this one so I create a new one. 👍

**Comment by siddhpant** [comment]
Sometimes it also says no messages to show (or something like that).

**Comment by borisyankov** [comment]
@SerekKiri we are working on an issue that is similar to this one but not exactly the same. 

Can you answer these to point us in the right direction:
* When this happens, do you enter the app through a notification? 
* By 'not loading' do you mean you are seeing a message loading indication but the messages do not appear or are you seeing messages but they are the old ones and you are not seeing the new ones?

**Comment by borisyankov** [comment]
The last comment you just did matches the issue we are fixing in #3151

**Comment by SerekKiri** [comment]
It's happening by entering app by notifications and just opening it too.
By 'not loading' I mean that I can see old messages, but the new one are loading only after pressing "Mark as read".

**Comment by siddhpant** [comment]
@borisyankov Not quite. The PR is about opening the app through notification.

Here, I have opened manually, no notifications and no new messages (but it primarily happens when there are new messages).

![Message not shown](https://i.postimg.cc/T3G9KB0M/Screenshot-2018-11-28-15-35-17-698-com-zulipmobile.png)

The language is Hindi, but it's the same thing.

**Comment by gnprice** [comment]
@siddhpant The issue you're describing is #3162 . I just edited the title of @borisyankov 's #3151, which fixes it, to reflect that it's broader than just from notifications :).

For any further discussion of that issue, let's use that issue thread and PR thread.
---

@SerekKiri Thanks for the report! I think this is an issue we haven't seen before.

The fact that hitting "Mark as read" resolves the issue seems like a very informative clue, but it's puzzling, hmm; I'm not immediately thinking of a mechanism by which that would help.

One thought:

* Is it possible that the new messages are loaded, but the window just isn't scrolled to them? If instead of hitting "Mark as read", you try scrolling down, does anything happen?

  That doesn't seem likely from your description, but just want to confirm and rule it out.

**Comment by SerekKiri** [comment]
Yes, I'm trying to scoll down. It seems like loading but nothing happend (I was waiting 5 minutes with internet connection ofc) and it's load only after clicking "Mark as read".

**Comment by siddhpant** [comment]
@gnprice @borisyankov Here's how it happens... https://imgur.com/a/P36fWuR

Sorry, had to censor out the chats, hope you understand that.
